packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/dhtpeer_test.go

```diff
@@ -279,22 +330,1073 @@ func TestRoutingStarFullConnectivity(t *testing.T) {
 	time.Sleep(1 * time.Second)
 	for i := range peers {
 		for j := range peers {
-			if i == j {
+			from := len(peers) - 1 - i
+			target := j
+
+			// Should be able to route to self though
+			if from == target {
 				continue
 			}
 
-			err := peers[i].RouteEnvelope(aea.Envelope{
-				To:      AgentsTestAddresses[j],
-				Sender:  AgentsTestAddresses[i],
+			err := peers[from].RouteEnvelope(&aea.Envelope{
+				To:      AgentsTestAddresses[target],
+				Sender:  AgentsTestAddresses[from],
 				Message: []byte("ping"),
 			})
 
 			if err != nil {
-				t.Error("Failed to RouteEnvelope from ", i, "to", j)
+				t.Error("Failed to RouteEnvelope from ", from, "to", target)
 			}
 
-			expectEnvelope(t, rxs[j])
-			expectEnvelope(t, rxs[i])
+			expectEnvelope(t, rxs[target])
+			expectEnvelope(t, rxs[from])
 		}
 	}
 }
+
+/*
+	DHT network: DHTClient
```

Comment:
Tests different patterns of `DHTClient` to {`DHTPeer`, `DHTClient`} communications